### DISCUSSION TITLE
How do I get started with Node.js

### DISCUSSION THREAD

**Discussion by ansiace** [sindresorhus/awesome-nodejs#1292] [Q&A] [answered]
Are there any good resources to get started with Node.JS? Any good tutorials, blogs or books?

Of course, I have visited its official website http://nodejs.org/, but I didn't think the documentation they have is a good starting point.

**Comment by mdazfar2** [comment, marked as answer]
**@ansiace You can follow these tutorials to get started:**

### Tutorials-
- [NodeSchool.io interactive lessons](http://nodeschool.io/)

- [The Art of Node (an introduction to Node.js)](https://github.com/maxogden/art-of-node/#the-art-of-node)

- [Hello World](http://www.nodebeginner.org/#hello-world)

- [Hello World Web Server](http://www.nodebeginner.org/#building-the-application-stack) (paid)

- [Node.js guide](http://nodeguide.com/)

- [Build a blog with Node.js, express and MongoDB](http://howtonode.org/express-mongodb)

- [Node.js for Beginners](http://net.tutsplus.com/tutorials/javascript-ajax/node-js-for-beginners/)

- [Learn Node.js Completely and with Confidence](http://javascriptissexy.com/learn-node-js-completely-and-with-confidence/)

- [Node JS Processing Model – Single Threaded Model with Event Loop Architecture](http://www.journaldev.com/7462/node-js-processing-model-single-threaded-model-with-event-loop-architecture)

- [RisingStack's Node Hero Series](https://blog.risingstack.com/node-hero-tutorial-getting-started-with-node-js/)

- [Great Node.js tutorials voted by the programming community](https://hackr.io/tutorials/learn-node-js)

- [Node.js Tutorial](https://www.tutorialkart.com/nodejs/nodejs-tutorial/)

- [30 days of Node](https://www.nodejsera.com/30-days-of-node.html)

### Developer Sites-

- [Joyent's developer site for node](http://www.joyent.com/developers/node)
- [Tutorials Teacher](https://www.tutorialsteacher.com/nodejs/nodejs-tutorials)

### Videos-

- [Node Tuts](http://nodetuts.com/) (Node.js video tutorials)
- [Einführung in Node.js (in German)](https://vimeo.com/thenativeweb)
- [Introduction to Node.js with Ryan Dahl](http://www.youtube.com/watch?v=jo_B4LTHi3I)
- [Node.js: Asynchronous Purity Leads to Faster Development](http://www.infoq.com/presentations/nodejs)
- [Parallel Programming with Node.js](http://www.infoq.com/presentations/Parallel-Programming-with-Nodejs)
- [Server-side JavaScript with Node, Connect & Express](http://vimeo.com/18077379)
- [Node.js First Look](http://www.lynda.com/Nodejs-tutorials/Nodejs-First-Look/101554-2.html)
- [Node.js with MongoDB](http://www.youtube.com/watch?v=0_GNHWZHc-o)
- [Ryan Dahl's Google Tech Talk](http://www.youtube.com/watch?v=F6k8lTrAE2g)
- [Real Time Web with Node.js](http://node.codeschool.com/levels/1)
- [Node.js Tutorials for Beginners](https://www.youtube.com/playlist?list=PL6gx4Cwl9DGBMdkKFn3HasZnnAqVjzHn_)
- [Pluralsight courses (paid)](http://www.pluralsight.com/search/?searchTerm=Node.js)
- [Udemy Learn and understand Nodejs (paid)](https://www.google.co.in/url?sa=t&rct=j&q=&esrc=s&source=web&cd=1&cad=rja&uact=8&ved=0ahUKEwj6oMvR35nTAhWFuY8KHfC-BFoQFggZMAA&url=https%3A%2F%2Fwww.udemy.com%2Funderstand-nodejs%2F&usg=AFQjCNF3WkS8cFyYrTgzgd5V_S1ANKkZJw&sig2=rD4t8JYDPvQKjwAMzQnH7Q)
- [The New Boston](https://www.youtube.com/watch?v=-u-j7uqU7sI&list=PL6gx4Cwl9DGBMdkKFn3HasZnnAqVjzHn_)

### Screencasts-

- [Learn All The Nodes](https://www.youtube.com/playlist?list=PLQmX5gaHg2SXyKuT9BQ_nbFIdZ39yeRxH)
- [Node Tuts](http://nodetuts.com/)
- [Einführung in Node.js (in German)](https://vimeo.com/thenativeweb)
- [NodeCasts](http://nodecasts.net/)

### Books-

- [The Node Beginner Book](http://nodebeginner.org/)
- [Mastering Node.js](https://github.com/tj/masteringnode)
- [Up and Running with Node.js](http://chimera.labs.oreilly.com/books/1234000001808/index.html)
- [Node.js in Action](http://www.manning.com/cantelon/)
- [Smashing Node.js: JavaScript Everywhere](https://rads.stackoverflow.com/amzn/click/com/B008Z5OEUY)
- [Node.js & Co. (in German)](http://www.amazon.de/dp/389864829X)
- [Sam's Teach Yourself Node.js in 24 Hours](http://nodejsbook.io/)
- [Most detailed list of free JavaScript Books](http://jsbooks.revolunet.com/)
- [Mixu's Node Book](http://book.mixu.net/node/index.html)
- [Node.js the Right Way: Practical, Server-Side JavaScript That Scale](http://pragprog.com/book/jwnode/node-js-the-right-way)
- [Beginning Web Development with Node.js](https://leanpub.com/webdevelopmentwithnodejs)
- [Node Web Development](http://www.packtpub.com/node-javascript-web-development/book)
- [NodeJS for Righteous Universal Domination!](http://node.codeschool.com/)

### Courses

- [Real Time Web with Node.js](http://node.codeschool.com/)
- [Essential Node.js from DevelopMentor](http://www.develop.com/training-course/nodejs-featuring-node-npm-express-mocha-mongodb-with-mongoose)
- [Freecodecamp - Learn to code for free](https://www.freecodecamp.org/)
- [Udemy - The Complete Node.js Developer Course (3rd Edition) (paid)](https://www.udemy.com/course/the-complete-nodejs-developer-course-2/)

### Blogs-

- [The Node.js blog](http://blog.nodejs.org/)
- [How To Node](http://howtonode.org/)
- [DailyJS](http://dailyjs.com/)
- [Nodejitsu blog](http://blog.nodejitsu.com/)
- [Ryan Wilcox's Whitepaper](http://www.wilcoxd.com/whitepapers/node_js/)
- [devthought](http://www.devthought.com/)

### Podcasts-

- [NodeUp](http://nodeup.com/)

### JavaScript resources-

- [Crockford's videos](https://www.youtube.com/playlist?list=PL7664379246A246CB) (must see!)
- [Essential JavaScript Design Patterns For Beginners](http://www.addyosmani.com/resources/essentialjsdesignpatterns/book/)
- [JavaScript garden](http://bonsaiden.github.com/JavaScript-Garden/)
- [JavaScript Patterns](http://oreilly.com/catalog/9780596806767) book
- [JavaScript: The Good Parts](http://oreilly.com/catalog/9780596517748/) book
- [Eloquent javascript ](http://eloquentjavascript.net/)book

### Node.js Modules-

- [Search for registered Node.js modules](http://npmjs.org/)
- [A curated list of awesome Node.js libraries](https://github.com/sindresorhus/awesome-nodejs)
- [Wiki List on GitHub/Joyent/Node.js](https://github.com/joyent/node/wiki/modules) (start here last!)

### Other- 

- [JSApp.US - like jsfiddle, but for Node.js](http://jsapp.us/)
- [Node with VJET JS (for Eclipse IDE)](https://www.ebayopensource.org/index.php/VJET/NodeJS)
Production sites with published source:
- [Node Knockout Hackathon](http://nodeknockout.com/) [(source)](https://github.com/nko3/website)
- [Freecodecamp - Learn to code for free](https://www.freecodecamp.org/) [(source)](https://github.com/freeCodeCamp/freeCodeCamp)
- [Useful Node.js Tools, Tutorials and Resources](http://coding.smashingmagazine.com/2011/09/16/useful-node-js-tools-tutorials-and-resources/)
- [Runnable.com - like jsfiddle, but for server side as well](http://runnable.com/)
- [Getting Started with Node.js on Heroku](https://devcenter.heroku.com/categories/nodejs)
- [Getting Started with Node.js on Open-Shift](https://blog.openshift.com/run-your-nodejs-projects-on-openshift-in-two-simple-steps/)
- [Authentication using Passport](http://passportjs.org/guide/)
